File: services/policy-service/Models/Policy.cs
using System.ComponentModel.DataAnnotations;

namespace PolicyService.Models
{
    public class Policy
    {
        [Key]
        public Guid PolicyId { get; set; } = Guid.NewGuid();
        [Required]
        public string Title { get; set; } = default!;
        public string? Description { get; set; }
        public string? OwnerId { get; set; }
        public int ReviewCycleDays { get; set; } = 365;
        public string Status { get; set; } = "Active";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/web-portal/src/app/services/policy.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Policy { policyId?: string; title:string; description?:string; ownerId?:string; reviewCycleDays?:number; status?:string; }

@Injectable({ providedIn: 'root' })
export class PolicyService {
  private api = 'https://localhost:7243/api/v1/policies'; // dev URL, update when deployed
  constructor(private http: HttpClient) {}
  list(): Observable<Policy[]> { return this.http.get<Policy[]>(this.api); }
  create(p: Policy) { return this.http.post<Policy>(this.api, p); }
  update(id:string, p: Policy) { return this.http.put(`${this.api}/${id}`, p); }
  delete(id:string) { return this.http.delete(`${this.api}/${id}`); }
} 
